### PR TITLE
Prevent interval start > end in revertLocalSequenceRemove

### PR DESCRIPTION
**Pull request by scarlettjlee**
In revertLocalSequenceRemove, it's possible to revert interval positions such that start > end, which is probably not the intent of the undo. This can happen if only one endpoint was in the deleted text range, and then another client moved the interval in the meantime. Or if the interval was already reversed. In this change, it won't move the interval if that will result in start > end.

I tried to prevent start > end in IntervalCollection.change, but it broke 126 tests (mostly fuzz) and fixing all of them was taking longer than a day. So I'm tabling that for now.

### CODE REVIEW

File: packages/dds/sequence/src/revertibles.ts [hunk 1]
```diff
@@ -453,8 +453,9 @@ function revertLocalChange(
 	const startSlidePos = getSlidePosition(string, revertible.start, start);
 	const end = string.localReferencePositionToPosition(revertible.end);
 	const endSlidePos = getSlidePosition(string, revertible.end, end);
-	if (!isValidRange(startSlidePos, endSlidePos, string)) return;
-	collection.change(id, startSlidePos, endSlidePos);
+	if (isValidRange(startSlidePos, endSlidePos, string)) {
+		collection.change(id, startSlidePos, endSlidePos);
+	}
```
Comment (scarlettjlee) [inline packages/dds/sequence/src/revertibles.ts:458]: This change is not related to the rest of the PR. I happened to notice that revertible local refs aren't being cleaned up in this case.

**Review by Abe27342 — APPROVED** [review]

File: packages/dds/sequence/src/revertibles.ts [hunk 2]
```diff
@@ -545,7 +546,16 @@ function revertLocalSequenceRemove(
 				restoredRanges,
 				sharedString,
 			);
-			if (newStart !== undefined || newEnd !== undefined) {
+			// only move interval if start <= end
+			if (
```
Comment (Abe27342) [inline packages/dds/sequence/src/revertibles.ts:550]: might be nice if you could do something like `(newStart ?? sharedString.localReferencePositionToPosition(interval.start)) <= (newEnd ?? sharedString.localReferencePositionToPosition(interval.end))`. but it's not obvious to me that at least one of newStart/newEnd is undefined. if not, I think this is better anyway.

File: packages/dds/sequence/src/revertibles.ts (hunk 1 above)
Comment (jzaffiro) [reply packages/dds/sequence/src/revertibles.ts:458]: This makes more sense than what I had before anyway. I probably had something similar in revertLocalDelete, so if you want to change that instance as well, go ahead. If not, I can make the change as part of [AB#4859](https://dev.azure.com/fluidframework/internal/_workitems/edit/4859).

**Review by jzaffiro — APPROVED** [review]

File: packages/dds/sequence/src/revertibles.ts (hunk 2 above)
Comment (scarlettjlee) [reply packages/dds/sequence/src/revertibles.ts:550]: They could both be undefined.
